### ISSUE TITLE
Rename cosine distance with cosine similarity

### ISSUE THREAD

**Issue by hahahannes** [pytorch/pytorch#75104]
### 📚 The doc issue

The documentation at [https://pytorch.org/docs/stable/generated/torch.nn.CosineEmbeddingLoss.html](https://pytorch.org/docs/stable/generated/torch.nn.CosineEmbeddingLoss.html) states that the loss function uses the `cosine distance` while actually implementing the loss function using the `cosine similarity`.  

### Suggest a potential alternative/fix

Maybe renaming `cosine distance` to `cosine similarity` would be more concise.

cc @brianjo @mruberry @albanD @jbschlosser @walterddr @kshitij12345

**Comment by albanD** [comment]
Hi,
Thanks for the report, could you send a small PR fixing that?
